### ISSUE TITLE
Duplicate module name error

### ISSUE THREAD

**Issue by manikantag** [esbenp/react-native-clean-form#8]
Hi,

I m getting **Duplicate module name error** errors *(see below for actual error)* when I've installed `react-native-clean-form`. If I remove this error is not coming. With little Google search, I think library needs to include `react` and `react-native` as `peerDependency`.

See below:
https://github.com/RealOrangeOne/react-native-mock/issues/58
https://github.com/stephy/CalendarPicker/issues/24
https://github.com/facebook/react-native/issues/3558
https://github.com/facebook/react-native/issues/7384

```
...
This warning is caused by a @providesModule declaration with the same name across two different files.
jest-haste-map: @providesModule naming collision:
  Duplicate module name: ReactHostOperationHistoryHook
  Paths: C:\Users\kh1015\manikanta\ditcos-work\readymix\node_modules\react-native-clean-form\node_modules\react-native\Libraries\Renderer\src\renderers\shared\hooks\ReactHostOperationHistoryHook.js collides with C:\Users\kh1015\manikanta\ditcos-work\readymix\node_modules\react-native\Libraries\Renderer\src\renderers\shared\hooks\ReactHostOperationHistoryHook.js

This warning is caused by a @providesModule declaration with the same name across two different files.
jest-haste-map: @providesModule naming collision:
  Duplicate module name: ReactInstanceMap
  Paths: C:\Users\kh1015\manikanta\ditcos-work\readymix\node_modules\react-native-clean-form\node_modules\react-native\Libraries\Renderer\src\renderers\shared\shared\ReactInstanceMap.js collides with C:\Users\kh1015\manikanta\ditcos-work\readymix\node_modules\react-native\Libraries\Renderer\src\renderers\shared\shared\ReactInstanceMap.js

This warning is caused by a @providesModule declaration with the same name across two different files.
jest-haste-map: @providesModule naming collision:
  Duplicate module name: shouldUpdateReactComponent
  Paths: C:\Users\kh1015\manikanta\ditcos-work\readymix\node_modules\react-native-clean-form\node_modules\react-native\Libraries\Renderer\src\renderers\shared\shared\shouldUpdateReactComponent.js collides with C:\Users\kh1015\manikanta\ditcos-work\readymix\node_modules\react-native\Libraries\Renderer\src\renderers\shared\shared\shouldUpdateReactComponent.js

This warning is caused by a @providesModule declaration with the same name across two different files.
jest-haste-map: @providesModule naming collision:
  Duplicate module name: ReactErrorUtils
  Paths: C:\Users\kh1015\manikanta\ditcos-work\readymix\node_modules\react-native-clean-form\node_modules\react-native\Libraries\Renderer\src\renderers\shared\utils\ReactErrorUtils.js collides with C:\Users\kh1015\manikanta\ditcos-work\readymix\node_modules\react-native\Libraries\Renderer\src\renderers\shared\utils\ReactErrorUtils.js

This warning is caused by a @providesModule declaration with the same name across two different files.
jest-haste-map: @providesModule naming collision:
  Duplicate module name: ReactTypeOfWork
  Paths: C:\Users\kh1015\manikanta\ditcos-work\readymix\node_modules\react-native-clean-form\node_modules\react-native\Libraries\Renderer\src\renderers\shared\fiber\ReactTypeOfWork.js collides with C:\Users\kh1015\manikanta\ditcos-work\readymix\node_modules\react-native\Libraries\Renderer\src\renderers\shared\fiber\ReactTypeOfWork.js
...
```

**Comment by dekm** [comment]
I see this as well.  It must be an issue with the latest react-native update.

**Comment by dekm** [comment]
I deleted the node_modules folder inside of react-native-clean-form and the errors are gone.  Not sure what kind of dependencies it has on that.  It appears to just be a duplicate of react-native and some other common packages though.

**Comment by manikantag** [comment]
In fact, I've deleted the `react-native` from `node_modules/react-native-clean-form/node_modules`

**Comment by weiliank** [comment]
@manikantag do you have this issue mentioned in #7 after deleting the `react-native` folder?

**Comment by manikantag** [comment]
@weiliank I m not using `redux-form`, yet!

**Comment by mwarger** [comment]
@manikantag npm start -- --reset-cache fixed this for me.

**Comment by manikantag** [comment]
@mwarger Oh, thanks. Will try that,

**Comment by markusguenther** [comment]
These issues are related https://github.com/esbenp/react-native-clean-form/issues/6

I created a small pull request https://github.com/esbenp/react-native-clean-form/pull/12 to fix that.

**Comment by markusguenther** [comment]
@esbenp this can be closed after #12 has been merged ;)

**Comment by markusguenther** [comment]
Thanks for reporting. No feedback at the moment and with the current version I can not reproduce this issue.

If there are still issues please reopen the issue and report your problems :)

**Comment by lionhive** [comment]
I am having same problem, detailed steps here:
https://github.com/esbenp/react-native-clean-form/issues/36
